src/main/java/org/springframework/data/repository/query/QueryMethod.java

```diff
@@ -86,7 +86,7 @@ public QueryMethod(Method method, RepositoryMetadata metadata, ProjectionFactory
 		this.method = method;
 		this.unwrappedReturnType = potentiallyUnwrapReturnTypeFor(metadata, method);
 		this.metadata = metadata;
-		this.parameters = createParameters(method, metadata.getDomainTypeInformation());
```

Comment:
This one needs to stay as-is to ease compatibility. We cannot assume that all stores have implemented `createParameters(ParametersSource parametersSource)`.